CIP-?/README.md

```diff
@@ -0,0 +1,718 @@
+
+---
+```haskell
+iorByteString :: BuiltinByteString -> BuiltinByteString -> BuiltinByteString
+```
+Perform a bitwise logical IOR on arguments of the same
+length, producing a result of the same length, erroring otherwise.
+
+---
+```haskell
+xorByteString :: BuiltinByteString -> BuiltinByteString -> BuiltinByteString
+```
+Perform a bitwise logical XOR on arguments of the same
+length, producing a result of the same length, erroring otherwise.
+
+---
+```haskell
+complementByteString :: BuiltinByteString -> BuiltinByteString
+```
+Complement all the bits in the argument, producing a
+result of the same length.
+
+
+---
+Lastly, we define the following additional operations:
+
+
+```haskell
+shiftByteString :: BuiltinByteString -> BuiltinInteger -> BuiltinByteString
+```
+Performs a bitwise shift of the first argument by the
+absolute value of the second argument, with padding, the direction being
+indicated by the sign of the second argument.
+
+---
+```haskell
+rotateByteString :: BuiltinByteString -> BuiltinInteger -> BuiltinByteString
+```
+Performs a bitwise rotation of the first argument by
+the absolute value of the second argument, the direction being indicated by
+the sign of the second argument.
+
+---
+```haskell
+popCountByteString :: BuiltinByteString -> BuiltinInteger
+```
+Returns the number of $1$ bits in the argument.
+
+---
+```haskell
+testBitByteString :: BuiltinByteString -> BuiltinInteger -> BuiltinBool
+```
+If the position given by the second argument is not in
+bounds for the first argument, error; otherwise, if the bit given by that
+position is $1$, return `True`, and `False` otherwise.
+
+---
+```haskell
+writeBitByteString :: BuiltinByteString -> BuiltinInteger -> BuiltinBool -> BuiltinByteString
+```
+If the position given by the second
+argument is not in bound for the first argument, error; otherwise, set the
+bit given by that position to $1$ if the third argument is `True`,
+and $0$ otherwise.
+
+---
+```haskell
+findFirstSetByteString :: BuiltinByteString -> BuiltinInteger
+```
+Return the lowest index such that `testBitByteString` with the first
+argument and that index would be `True`. If no such index exists,
+return $-1$ instead.
+
+
+## Semantics
+
+### Preliminaries
+
+We define $\mathbb{N}^{+} = \{ x \in \mathbb{N} \mid x \neq 0 \}$. We assume
+that `BuiltinInteger` is a faithful representation of $\mathbb{Z}$. A
+**bit sequence** $s = s_n s_{n-1} \ldots s_0$ is a sequence such that for
+all $i \in \{0,1,\ldots,n\}$, $s_i \in \{0, 1\}$. A bit sequence $s = s_n s_{n-1} \ldots s_0$ is a **byte sequence** if $n = 8k - 1$ for some $k \in \mathbb{N}$. We denote the **empty bit sequence** (and, indeed, byte sequence
+as well) by $\emptyset$.
+
+We intend that `BuiltinByteString`s represent byte sequences, with the
+sequence of bits being exactly as the description above. For example, given the
+byte sequence `0110111100001100`, the `BuiltinByteString`
+corresponding to it would be `o\f`.
```

Comment:
It would be under the encoding we describe in this section. How could I phrase this more clearly?

Reply:
You don't describe a _string_ encoding here. I guessed that it might be ASCII and it looks like it is "The bytestring corresponding to it would be the one corresponding to the ASCII encoding of `o\f`"